### DISCUSSION TITLE
Running a python script from Macro or a Plugin?

### DISCUSSION THREAD

**Discussion by mcinquino** [winder/Universal-G-Code-Sender#1688] [Ideas]
Hello,

I have a system with a Raspberry Pi 4 running UGS Platform talking to GRBL v1.1. Arduino Uno is the  GRBL hardware. I am looking for a way to more seamlessly tie E-stop/Spindle Power and Stepper drive power to the status of UGS. I have noticed that on power up of the Raspberry Pi I get chatter on the spindle direction relay. I am using 0-10V speed control on a VFD. I am concerned that if I am getting unexpected chatter of that relay on startup that I might also get other unexpected behavior. My thought is to prevent any unexpected motion or spindle activity by not allowing power to them until UGS is up and running. On a Raspberry Pi I could do this fairly easily if I could control an output from within UGS. If I could launch a script lets say from a Macro button or a plugin tab I could require it to enable the stepper power and spindle power. Has anyone done anything similar to this? Would a plugin or some method to launch add on apps or scripts be useful for other things? I could see this being much easier on a Raspberry Pi because of the IO available vs a windows system. Knowing if an estop was pressed or a specific drive was faulted would be handy. Having the ability to tie non critical machine status information back to the GUI would really make for some nice systems. Any thoughts on something like this?

**Comment by breiler** [comment]
Similar requests have been made before about running a script on a certain event that would maybe solve this. But such a feature is currently low on my priority list =(:
https://github.com/winder/Universal-G-Code-Sender/issues/239

Regarding the chatter when powering up sounds like a common issue described here:
https://github.com/gnea/grbl/wiki/Frequently-Asked-Questions#my-spindle-turns-on-when-i-boot-up-my-arduino-for-a-second-or-two-why-does-it-do-this

**Comment by mcinquino** [comment]
I understand. I may give a try at making a plugin. I think my biggest challenge will be getting the development environment setup and getting things to compile after making it. 

Thanks for the link on the chatter. I had not found that. The suggested solutions look like good options.
